docs/transform.ipynb

```diff
@@ -116,61 +410,90 @@
+    "\n",
+    "The post-process transformers often includes cleanups and optimizations like dropping negligible operations, converting single qubit rotations into desired form, circuit alignments etc. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "-QFPCdW0qS3R"
+   },
+   "outputs": [],
+   "source": [
+    "context = cirq.TransformerContext(logger=cirq.TransformerLogger())\n",
+    "converted_circuit_optimized = cirq.optimize_for_target_gateset(circuit, gateset = cirq.CZTargetGateset(allow_partial_czs=True), context=context)\n",
+    "cirq.testing.assert_circuits_with_terminal_measurements_are_equivalent(circuit, converted_circuit_optimized)\n",
+    "print(f\"Circuit Depth Original: {len(circuit)}\", circuit, sep=\"\\n\")\n",
+    "print(f\"Circuit Depth Converted Custom: {len(converted_circuit)}\", converted_circuit, sep=\"\\n\")\n",
+    "print(f\"Circuit Depth Converted Optimized: {len(converted_circuit_optimized)}\", converted_circuit_optimized, sep=\"\\n\")"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "id": "e046ef24c70e"
+    "id": "CmJ7aYz7riRT"
    },
    "outputs": [],
    "source": [
-    "class RemoveMeasurements(cirq.PointOptimizer):\n",
-    "    def optimization_at(self, circuit: cirq.Circuit, index: int, op: cirq.Operation):\n",
-    "        if isinstance(op.gate, cirq.MeasurementGate):\n",
-    "            return cirq.PointOptimizationSummary(clear_span=1,\n",
-    "                                                 new_operations=[],\n",
-    "                                                 clear_qubits=op.qubits)\n",
-    "        else:\n",
-    "            return None\n",
-    "\n",
-    "q=cirq.LineQubit(0)\n",
-    "c=cirq.Circuit(cirq.X(q), cirq.measure(q))\n",
-    "print('Before optimization')\n",
-    "print(c)\n",
-    "RemoveMeasurements().optimize_circuit(c)\n",
-    "print('After optimization')\n",
-    "print(c)\n"
+    "print(\"Inspecting actions of individual transformers when running `cirq.optimized_for_target_gateset`\\n\\n\")\n",
+    "context.logger.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "aTNKFcPesNpy"
+   },
+   "source": [
+    "# Summary\n",
+    "Cirq provides a flexible and powerful framework to\n",
+    "* Use built-in transformer primitives and analytical tools to create powerful custom transformers AND \n",
+    "* Easily integrate custom transformers with built-in infrastructure to augment functionality like automated logging, recursive execution on sub-circuits, support for no-compile tags etc.\n",
+    "\n",
+    "Cirq also provides a plethora of built-in transformers which can be composed together into a sensible abstraction, like `cirq.CompilationTargetGateset`, which in-turn can be serialized and can be used as a parameter in larger compilation pipelines and experiment workflows. "
```

Comment:
Usually we want to end on some kind of "Now you go and build on this for yourself". Could we rework this into something like: "Try using transformers to optimize your circuits for your compilation needs" or "try adapting the above transformers to work with SycamoreGates" or something ?

Reply:
Added an ending line with link to api reference docs.